### PR TITLE
Various clean-ups before assert adjustment

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

Well, I've ended up collecting some cleanups I'd like to settle down before #12671.

#### Summary of Changes

Flush out various cleanups before really introducing assert adjustments for easier review.

This pr should be no functional change.

### CODE REVIEW

File: core/src/consensus.rs [hunk 1]
```diff
@@ -860,30 +863,24 @@ impl Tower {
             )
         );
 
-        // return immediately if votes are empty...
-        if self.lockouts.votes.is_empty() {
-            return Ok(self);
```
Comment (ryoqun) [inline core/src/consensus.rs:865]: I didn't like my previous coding (especially regarding the branched-off early bail-out code flow with `Ok(_)`)...  Well, I found out that I can steamline these a bit.

File: core/src/consensus.rs [hunk 2]
```diff
@@ -931,7 +928,12 @@ impl Tower {
                 if !voting_from_genesis {
                     // Unless we're voting since genesis, slots_in_tower must always be older than last checked_slot
                     // including all vote slot and the root slot.
-                    assert!(*slot_in_tower < checked_slot)
+                    assert!(
```
Comment (ryoqun) [inline core/src/consensus.rs:931]: I'm adding tests before adjusting this assert!.

File: core/src/consensus.rs [hunk 3]
```diff
@@ -2999,4 +3000,54 @@ pub mod test {
             "The tower is fatally inconsistent with blockstore: not too old once after got too old?"
         );
     }
+
+    #[test]
```
Comment (ryoqun) [inline core/src/consensus.rs:3004]: I'm adding tests before adjusting this assert!.

File: core/src/consensus.rs [hunk 4]
```diff
@@ -1176,7 +1177,7 @@ pub fn reconcile_blockstore_roots_with_tower(
             "at least 1 parent slot must be found"
         );
 
-        blockstore.set_roots(&new_roots)?
+        blockstore.set_roots(&new_roots)?;
```
Comment (ryoqun) [inline core/src/consensus.rs:1180]: oddly enough, clippy and rustfmt doesn't seem to complain this. ;)

**Comment by ryoqun** [comment]
@carllin Could you review this? Thanks!

File: core/src/consensus.rs [hunk 5]
```diff
@@ -860,30 +863,24 @@ impl Tower {
...
+            if slot_history.check(last_voted_slot) == Check::TooOld {
+                // We could try hard to anchor with other older votes, but opt to simplify the
+                // following logic
+                return Err(TowerError::TooOldTower(
+                    last_voted_slot,
+                    slot_history.oldest(),
+                ));
+            }
+            self.adjust_lockouts_with_slot_history(slot_history)?;
+            self.initialize_root(replayed_root);
```
Comment (carllin) [inline core/src/consensus.rs:876]: cool, I think this looks equivalent to me, much easier to read 😄

**Review by carllin — APPROVED** [review]

File: core/src/consensus.rs (hunk 2 above)
Comment (xni) [reply core/src/consensus.rs:931]: Hi! I'm catching this assertion when I'm running a new validator: https://github.com/solana-labs/solana/issues/13128

**Pull request merged**